packages/core/src/node/messaging/ipc-channel.ts

```diff
@@ -49,12 +48,13 @@ export class IPCChannel implements Channel {
 
     protected ipcErrorListener: (error: Error) => void = error => this.onErrorEmitter.fire(error);
 
-    constructor(childProcess?: cp.ChildProcess) {
+    constructor(socket: Socket, childProcess?: cp.ChildProcess) {
         if (childProcess) {
             this.setupChildProcess(childProcess);
         } else {
             this.setupProcess();
         }
+        this.messagePipe = new BinaryMessagePipe(socket);
```

Comment:
Do we really need a `Socket`, or is `Duplex` enough here?

```suggestion
    constructor(connection: Duplex, childProcess?: cp.ChildProcess) {
        if (childProcess) {
            this.setupChildProcess(childProcess);
        } else {
            this.setupProcess();
        }
        this.messagePipe = new BinaryMessagePipe(connection);
```

Reply:
Duplex should be enough